hypervisor/dm/vpci/pci_pt.c

```diff
@@ -426,6 +407,10 @@ static void init_bars(struct pci_vdev *vdev, bool is_sriov_bar)
 				if (!is_sriov_bar) {
 					pci_vdev_write_vbar(vdev, idx, lo);
 				}
+
+				if (is_pci_io_bar(vbar)) {
+					vdev_pt_allow_io_vbar(vdev, idx);
+				}
```

Comment:
Should we check if the vBAR is same with pBAR before update VMCS's I/O bitmap?

Reply:
Yes, we will check it later.
Because acrn-config hasn't ensured PIO bar identical mapping now.
So we didn't check now.
Actually, besides this part, when guest reprograms the PIO bar, we need to check too.

Reply:
Even the acrn-config tool has kept the identical mapping, we still need to check here due to SOS is not trusted. Can you add check now? I think it expose another security issue.

Reply:
As we discussed with Eddie, We need to inject #GP to guest if not identical mapping.
Then if pre-launched VM pass-thru device that has PIO BAR, then it couldn't work.
I didn't know any security issue, it may impact the functionality here.

Reply:
I haven't notice the base_gpa has been changed to bas_hpa.. It is fine now

Reply:
ok